### PR TITLE
Arn 770 add multiple data from delius

### PR DESCRIPTION
**Pull request by wood-n**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/assessments/services/EpisodeService.kt [hunk 1]
```diff
@@ -107,31 +112,22 @@ class EpisodeService(
...
   private fun prepopulateFromSource(
     episode: AssessmentEpisodeEntity,
-    sourceName: String?,
-    questions: List<ExternalSourceQuestionSchemaDto>,
+    sourceName: String,
+    questionSchemas: List<ExternalSourceQuestionSchemaDto>,
     latestCompleteEpisodeEndDate: LocalDateTime?
   ) {
-    val questionsByExternalSourceEndpoint = questions.groupBy { it.externalSourceEndpoint }
-    episode.prepopulatedFromOASys = sourceName == OASYS_SOURCE_NAME
```
Comment (markheyward-chaplin) [inline src/main/kotlin/uk/gov/justice/digital/assessments/services/EpisodeService.kt:115]: episode.prepopulatedFromOASys = sourceName == OASYS_SOURCE_NAME needs to be added for RSR

File: src/main/resources/db/migration/refdata/V1_7__upw_assessment.sql [hunk 2]
```diff
@@ -216,6 +217,7 @@ VALUES
 ('19a1b4d8-0f85-4049-ba7b-127646c882d8', 'pregnancy_recently_given_birth_details', '2021-09-27 14:50:00', null, 'textarea', null, 'Give details', null, null),
 ('155e1596-9f83-4e0a-8fa7-e36f9ed2b2e5', 'health_issues_complete', '2021-09-27 14:50:00', null, 'radio', '8067ff6e-7400-4d1e-ae2a-87dee7e124ec', 'Mark this section as complete?', '', null),
 ('0ba4ef3e-8c40-4c1e-b267-a42afd07ee4e', 'gp_details_complete', '2021-09-27 14:50:00', null, 'radio', '8067ff6e-7400-4d1e-ae2a-87dee7e124ec', 'Mark this section as complete?', '', null),
+('2a1a41dd-5d77-4674-9f8f-561c654867b8', 'gp_details', '2021-09-27 14:50:00', null, 'table', null, '', '', null),
```
Comment (cairnsp) [inline src/main/resources/db/migration/refdata/V1_7__upw_assessment.sql:220]: Should we be changing existing flyway SQL files?

File: src/test/kotlin/uk/gov/justice/digital/assessments/services/AssessmentUpdateServiceTablesTest.kt [hunk 3]
```diff
@@ -1,358 +1,354 @@
...
+//
+//   @Test
+//   fun `adds an entry to a table for a given episode`() {
+//     val requestBody = UpdateAssessmentEpisodeDto(
+//       answers = mapOf(
+//         "first_question" to listOf("first_answer"),
+//         "second_question" to listOf("second_answer"),
+//       )
+//     )
+//
```
Comment (cairnsp) [inline src/test/kotlin/uk/gov/justice/digital/assessments/services/AssessmentUpdateServiceTablesTest.kt:93]: we should really commit commented out code to the repo -  could you please delete.

**Review by cairnsp — COMMENTED** [review]
Just a couple of minor comments

File: src/test/kotlin/uk/gov/justice/digital/assessments/services/AssessmentUpdateServiceTablesTest.kt (hunk 3 above)
Comment (wood-n) [reply src/test/kotlin/uk/gov/justice/digital/assessments/services/AssessmentUpdateServiceTablesTest.kt:93]: Thanks, had missed this file in the cleanup!

File: src/main/resources/db/migration/refdata/V1_7__upw_assessment.sql (hunk 2 above)
Comment (wood-n) [reply src/main/resources/db/migration/refdata/V1_7__upw_assessment.sql:220]: Normally, definitely not but this has been discussed and given a thumbs up in this instance as it's reference data and separate from the user data

File: src/main/kotlin/uk/gov/justice/digital/assessments/restclient/communityapi/PersonalContactDto.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  val notes: String?,
+  val gender: String?,
+  val relationshipType: KeyValue?,
+  val createdDatetime: LocalDateTime?,
+  val lastUpdatedDatetime: LocalDateTime?,
+  val address: AddressSummary?,
+  val isActive: Boolean?,
+)
+
+class KeyValue(
```
[the remark this answers is not in the document]
Comment (wood-n) [reply src/main/kotlin/uk/gov/justice/digital/assessments/restclient/communityapi/PersonalContactDto.kt:26]: This has been kept in line with the community api/delius naming as it's their dto. Happy to discuss though!

**Review by markheyward-chaplin — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/assessments/services/AssessmentService.kt [hunk 5]
```diff
@@ -252,7 +252,7 @@ class AssessmentService(
   }
 
   private fun matchAnswers(
-    episodeAnswer: Map.Entry<String, List<String>>,
+    episodeAnswer: Map.Entry<String, List<Any>>,
```
Comment (LucasCairns) [inline src/main/kotlin/uk/gov/justice/digital/assessments/services/AssessmentService.kt:255]: It's a shame Kotlin doesnt have `union`(?) types (AFAIK) - in TypeScript you would have something like
```
type AnswerValue = String | GPDetails | EmergencyContactDetails | PersonalContactDetails
```
Which would help remove the need for the `Any` type here

**Review by LucasCairns — APPROVED** [review]
LGTM 👍

File: src/main/resources/application-dev.yml [hunk 6]
```diff
@@ -6,7 +6,7 @@ server:
   forward-headers-strategy: native
 spring:
   flyway:
-    enabled: false
+    enabled: true
```
Comment (LucasCairns) [inline src/main/resources/application-dev.yml:9]: Will be interesting to see how this deploys to `dev`, does this mean we've not been running `Flyway`? 🤔

Comment (wood-n) [reply src/main/resources/application-dev.yml:9]: Nope! At least not in dev but prod is safe

**Pull request merged**